ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSource.mm

```diff
@@ -161,22 +171,20 @@ - (void)_enqueueModification:(id<CKTransactionalComponentDataSourceStateModifyin
 - (void)_startFirstAsynchronousModification
 {
   CKAssertMainThread();
-  id<CKTransactionalComponentDataSourceStateModifying> modification = _pendingAsynchronousModifications[0];
-  CKTransactionalComponentDataSourceState *baseState = _state;
-  dispatch_async(_workQueue, ^{
-    CKTransactionalComponentDataSourceChange *change = [modification changeFromState:baseState];
-    dispatch_async(dispatch_get_main_queue(), ^{
-      // If the first object in _pendingAsynchronousModifications is not still the modification,
-      // it may have been canceled; don't apply it.
-      if ([_pendingAsynchronousModifications firstObject] == modification && _state == baseState) {
-        [self _synchronouslyApplyChange:change];
-        [_pendingAsynchronousModifications removeObjectAtIndex:0];
-      }
-      if ([_pendingAsynchronousModifications count] != 0) {
-        [self _startFirstAsynchronousModification];
-      }
+  NSThread *currentWorkThread = _state.configuration.workThread;
```

Comment:
Pull the work thread from the current state's configuration. This way it's possible to change the work thread without creating a new transactional data source.